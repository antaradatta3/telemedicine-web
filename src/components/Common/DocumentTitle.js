import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

export const DocumentTitle = ({ prefix, listen }) => {
  const [pos, setPos] = useState({ x: 0, y: 0 });
  const [title, setTitle] = useState('');

  useEffect(() => {
    if (listen) {
      const handler = event => {
        setPos({ x: event.clientX, y: event.clientY });
      };
      window.addEventListener('mousemove', handler);

      return () => {
        window.removeEventListener('mousemove', handler);
      };
    }

    return () => null;
  }, [listen]);

  useEffect(() => {
    const positionTitle = `${prefix} - ${pos.x}, ${pos.y}`;
    document.title = positionTitle;
    setTitle(positionTitle);
  }, [pos, prefix]);

  return <h3>{title}</h3>;
};

DocumentTitle.propTypes = {
  prefix: PropTypes.string,
  listen: PropTypes.bool,
};

export default DocumentTitle;
