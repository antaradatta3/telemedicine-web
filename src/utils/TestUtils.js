import _toJson from 'enzyme-to-json';
import jestSnapshotDiff from 'snapshot-diff';

const toJson = data => _toJson(data, { noKey: true });

export const snapshotDiff = (component1, component2) => {
  return jestSnapshotDiff(toJson(component1), toJson(component2));
};
