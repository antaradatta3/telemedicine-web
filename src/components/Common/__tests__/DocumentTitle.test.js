import React from 'react';
import { mount } from 'enzyme';

import { snapshotDiff } from '../../../utils/TestUtils';
import DocumentTitle from '../DocumentTitle';

describe('Component: DocumentTitle', () => {
  const props = {
    prefix: 'demo',
    listen: false,
  };
  const component = mount(<DocumentTitle {...props} />);

  it('DocumentTitle matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('DocumentTitle matches snapshot Diff 1', () => {
    const component1 = mount(<DocumentTitle {...props} listen />);
    expect(snapshotDiff(component, component1)).toMatchSnapshot();
  });
});
