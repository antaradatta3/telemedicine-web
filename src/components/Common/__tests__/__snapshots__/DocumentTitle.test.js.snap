// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DocumentTitle DocumentTitle matches snapshot 1`] = `
<DocumentTitle
  listen={false}
  prefix="Sample"
>
  <h3>
    Sample - 0, 0
  </h3>
</DocumentTitle>
`;

exports[`Component: DocumentTitle DocumentTitle matches snapshot Diff 1 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <DocumentTitle
-   listen={false}
+   listen={true}
    prefix=\\"Sample\\"
  >
    <h3>
      Sample - 0, 0
    </h3>"
`;
